task/build-maven-zip/0.1/build-maven-zip.yaml

```diff
@@ -0,0 +1,194 @@
+    name: IMAGE_DIGEST
+  - description: OCI-Artifact repository and tag where the built OCI-Artifact was pushed
+    name: IMAGE_URL
+  - description: OCI-Artifact reference of the built OCI-Artifact
+    name: IMAGE_REF
+  - name: SBOM_BLOB_URL
+    description: Reference of SBOM blob digest to enable digest-based verification from provenance
+    type: string
+  stepTemplate:
+    env:
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: PKG_ROOT
+      value: $(params.PREFETCH_ROOT)
+    - name: FILE_NAME
+      value: $(params.FILE_NAME)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.IMAGE_EXPIRES_AFTER)
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+
+  steps:
+  - image: quay.io/konflux-ci/appstudio-utils@sha256:d0fecb67dd98d874704f2e3d20686363de7cbb42c15ad2b1d8e8c38118c50078
+    name: prepare
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
+    script: |
+      #!/bin/bash
+      set -euo pipefail
+      # Generate checksums for all maven artifact files. It will ignore the checksum files
+      # and signature files if they existed there
+      pkgpath="$(workspaces.source.path)/cachi2/output/deps/generic"
+      if [ -d "$pkgpath/${PKG_ROOT}" ]; then
+        echo "Generating checksums for artifacts"
+        while IFS= read -r -d '' f
+        do
+          md5sum "$f" | awk '{print $1}'> "$f.md5"
+          sha1sum "$f" | awk '{print $1}'> "$f.sha1"
+          sha256sum "$f" | awk '{print $1}'> "$f.sha256"
+        done <   <(find "$pkgpath/${PKG_ROOT}" -type f ! -name "*.md5" \
+        ! -name "*.sha1" ! -name "*.sha128" ! -name "*.sha256" \
+        ! -name "*.sha512" ! -name "*.asc" -print0)
+        # Bundle the artifacts and checksums together into a zip file
+        cd "$pkgpath"
+        echo "create maven zip to $(workspaces.source.path)/cachi2/output/${FILE_NAME}.zip"
+        zip -rq "${FILE_NAME}.zip" "${PKG_ROOT}"
+        mv "${FILE_NAME}.zip" "/shared/${FILE_NAME}.zip"
+      else
+        echo "No ${PKG_ROOT} dir found, can not generate maven zip!"
+        exit 1
+      fi
+    securityContext:
+      capabilities:
+        add:
+          - SETFCAP
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/konflux-ci/oras:latest@sha256:9d6db5840c70e65fefe041201cc7ffe2d1661bd0582b590b54787213ccfd76e9
+    name: build
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
+    script: |
+      #!/bin/bash
+      set -euo pipefail
+      if [ -f "/shared/${FILE_NAME}.zip" ]; then
+        mv "/shared/${FILE_NAME}.zip" "./${FILE_NAME}.zip"
+        select-oci-auth "$IMAGE" > auth.json
+        [ -n "$IMAGE_EXPIRES_AFTER" ] && EXPIRE_LABEL=("--annotation" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
+        oras push "$IMAGE" \
+          --registry-config auth.json \
+          "${EXPIRE_LABEL[@]}" \
```

Comment:
Won't the task always fail when  `IMAGE_EXPIRES_AFTER` is empty? (Because `EXPIRE_LABEL` will be unset here)

Reply:
Not sure. I will have a test with my user workspace.

Reply:
Just had a quick test which set the "image-expires-after" to "" in my test pipeline, seems it does not break anything. The oci-artifact is pushed to quay.io without any expire label.